pkg/process/checks/runner.go

```diff
@@ -0,0 +1,99 @@
+package checks
+
+import (
+	"errors"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+// RunnerConfig implements config for runners that work with CheckWithRealTime
+type RunnerConfig struct {
+	CheckInterval time.Duration
+	RtInterval    time.Duration
+
+	ExitChan       chan struct{}
+	RtIntervalChan chan time.Duration
+	RtEnabled      func() bool
+	RunCheck       func(options RunOptions)
+}
+
+type runnerWithRealTime struct {
+	RunnerConfig
+	ratio      int
+	counter    int
+	newTicker  func(d time.Duration) *time.Ticker
+	stopTicker func(t *time.Ticker)
```

Comment:
nit: why can't the `newTicker` be a `ticker` property of `runnerWithRealTime`? And why `stopTicker` has to be a property instead?

Reply:
Practically it's just dependency injection for unit tests, `ticker` is re-created on RT change in the goroutine, and stopping a ticker panics if it's not initialized (in cases when we inject it, it's not properly initialized, just the channel is initialized.